### PR TITLE
Add chessboard component

### LINKED ISSUE #15: Add Chess Board Component
Add a component to render the chess board.

### PR DESCRIPTION
**Pull request by codemayank**
Renders the chessboard
chessboard

resolves #15

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "chess.js": "https://github.com/jhlywa/chess.js#master",
```
Comment (nimish-gupta) [inline client/package.json:6]: why this?

File: client/src/components/chess-board/ChessBoard.js [hunk 2]
```diff
@@ -1,11 +1,95 @@
...
+class ChessBoard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      possibleMoves: [],
+      from: '',
+      prevSelectValidTouch: false,
+    };
+
+    this.handleSquareClick = this.handleSquareClick.bind(this);
```
Comment (nimish-gupta) [inline client/src/components/chess-board/ChessBoard.js]: You dont have to use this if you are using arrow functions in react component.

File: client/src/components/chess-board/ChessBoard.js [hunk 3]
```diff
@@ -1,11 +1,95 @@
...
-    ChessBoard Component
-    <ChessSquare />
-  </div>
-);
+const matchingStringInArray = array => str => array.some(x => x.includes(str));
+
+class ChessBoard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```
Comment (nimish-gupta) [inline client/src/components/chess-board/ChessBoard.js]: create them as class variables so that you don't have to use the constructor syntax

**Review by nimish-gupta — CHANGES_REQUESTED** [review]
LGTM after changes

File: client/package.json (hunk 1 above)
Comment (codemayank) [reply client/package.json:6]: There is no chess.board method on the chess.js library downloaded from npm so we had to pull the latest changes from Github. We need this method to update the board state after every move.

File: client/src/components/chess-board/ChessBoard.js (hunk 2 above)
Comment (codemayank) [reply client/src/components/chess-board/ChessBoard.js]: Done

File: client/src/components/chess-board/ChessBoard.js (hunk 3 above)
Comment (codemayank) [reply client/src/components/chess-board/ChessBoard.js]: Done

**Pull request merged**